mlir/include/mlir/Dialect/WebAssemblySSA/IR/WebAssemblySSAInterfaces.td

```diff
@@ -0,0 +1,186 @@
+//===-- WebAssemblySSAInterfaces.td - WebAssemblySSA Interfaces -*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines interfaces for the WebAssemblySSA dialect in MLIR.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef WEBASSEMBLYSSA_INTERFACES
+#define WEBASSEMBLYSSA_INTERFACES
+
+include "mlir/IR/OpBase.td"
+include "mlir/IR/BuiltinAttributes.td"
+
+def WasmSSALabelLevelInterface : OpInterface<"WasmSSALabelLevelInterface"> {
+  let description = [{
+    Operation that defines one level of nesting for wasm branching.
+    These operation region can be targeted by branch instructions.
```

Comment:
How do you guarantee the block is in the same region as the caller of the function?

Reply:
I suppose that by caller of the function you mean same region as the op, as this interface is intended for Wasm structured control flow operations (block, if(else) and. loop).

I've added a verifier for this (checking that either the op and target share the same parent region, or that the target is inside the op region).

For the existing op implementing this interface this is done "automatically" though: `if` and `block` returns their successor which is already verified, and `loop` returns its region entry block.